autogen.sh

```diff
@@ -6,27 +6,42 @@
 srcdir=`dirname $0`
 (test -d $srcdir/m4) || mkdir $srcdir/m4
 
-pushd $srcdir > /dev/null
-gtkdocize && \
-autoreconf --verbose --force --install --make || {
- echo 'autogen.sh failed';
- exit 1;
-}
-
-popd > /dev/null
-
-while test "x$@" != "x" ; do
-optarg=`expr "x$@" : 'x[^=]*=\(.*\)'`
```

Comment:
Removing this line seems like a possibly significant change in the option parsing. Is it necessary to remove it?